includes/payments/functions.php

```diff
@@ -1225,6 +1224,34 @@ function edd_get_next_payment_number() {
 }
 
 /**
+ * Given a prefix, number and postfix, return the payment number,  incremented
+ *
+ * @since  2.4
+ * @param  string $prefix  The Payment Number Prefix
+ * @param  string $number  The formatted Current Number to increment
+ * @param  string $postfix The Payment Number Postfix
+ * @return string          The new formatted Payment number
+ */
+function edd_increment_payment_number( $prefix, $number, $postfix ) {
+
+	// Remove prefix and postfix
+	$number = str_replace( $prefix, '', $number );
```

Comment:
Ok @chriscct7 yes, changing the prefix breaks the current functionality. But @pippinsplugins is right, we need to try and fix both the race condition and the prefix/postfix change issue. I'm going to close this PR, rewrite what I can, and we'll go from there.